html/modules/custom/ghi_menu/src/Form/MenuLinkDeleteWithSublinksForm.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Drupal\ghi_menu\Form;
+
+use Drupal\Core\Entity\ContentEntityDeleteForm;
+use Drupal\menu_link_content\Entity\MenuLinkContent;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Deletes a menu link with its Sublinks.
+ */
+class MenuLinkDeleteWithSublinksForm extends ContentEntityDeleteForm {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    $entity = $this->getEntity();
+    $config_name = MenuLinkDeleteWithSublinksSettingsForm::CONFIG_NAME;
+    $enabled_menus = \Drupal::config($config_name)->get('enabled_menus') ?? [];
```

Comment:
Use dependency injection for this. Calls to `\Drupal::` should not be done in non-static methods. You can use this type of pattern for simplicity: https://github.com/UN-OCHA/ghi-site/blob/d66d3cb4c765378daec564d024fe142c3e2f06cd/html/modules/custom/ghi_base_objects/src/Form/BaseObjectForm.php#L27